### ISSUE TITLE
Environment secrets appear to be empty when read in actions

### ISSUE THREAD

**Issue by bobthemighty** [integrations/terraform-provider-github#1257]
### Terraform Version
1.2.3

### Affected Resource(s)
github_actions_environment_secret

### Debug Output
No? It's full of secrets. It's literally me creating dozens of secrets. 

### Expected Behavior
Secrets created through Terraform should be usable from Github Actions

### Actual Behavior
Secrets are created in the environment, but seem to be empty when accessing them in Github Actions. Manually setting the secret in the environment fixes the issue.

The debug log shows a request to the secrets api, setting the value of `encrypted_value` correctly.
[github.tar.gz](https://github.com/integrations/terraform-provider-github/files/9353326/github.tar.gz)

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Create a secret in an environment through terraform
2. Read the value of the secret in github actions

### Important Factoids
There's a separate bug where Github Environments can't be created unless you have an enterprise account, since the provider tries to apply a branch protection rule, which fails for team accounts.
I worked around this issue by importing the environment into terraform state.

**Comment by bobthemighty** [comment]
When generating an `encrypted_value` string through `gh`, you need to specify the repository _and_ the environment that the secret is intended for. That makes it a bit more unwieldy to work with secrets across multiple environments, which was the reason I wanted to automate in the first place, but I can see the reasons why.

**Comment by shinenelson** [comment]
@bobthemighty, I got affected by this too. But I did not understand your explanation.

Looking at the state of a particular environment secret that I had created via terraform showed both the repository and environment values in the state.

```hcl
resource "github_actions_environment_secret" "audience" {
    created_at  = "2022-08-24 00:54:42 +0000 UTC"
    environment = "dev"
    id          = "[redacted-repository-name]:dev:audience"
    repository  = "[redacted-repository-name]"
    secret_name = "audience"
    updated_at  = "2022-08-24 00:54:42 +0000 UTC"
}
```

which has the repository and the environment in the state which implies to me that the secret is being created okay.

source code for creating the resource ( [as far back as 4.20.0](https://github.com/integrations/terraform-provider-github/blob/v4.20.0/github/resource_github_actions_environment_secret.go#L102) ) has passed the repository ID and the environment name along with the secret name in the API request.

https://github.com/integrations/terraform-provider-github/blob/bc38f5b42a142ecec8e8b338a47cc2d159a0ad53/github/resource_github_actions_environment_secret.go#L102

Is there something that I am missing here? Would you please be kind to elaborate a little more further?

**Comment by shinenelson** [comment]
oh, I was not generating the `encrypted_value` for terraform, I was directly adding `plaintext_value` ( which I was getting out of another encrypted file ). Let me try this other option out and see if it works for me as well.

**Comment by shinenelson** [comment]
My problem was not because of the provider. It was due to a bug in a module I was using.

I was naïve to hastily conclude that it was a problem with the provider as soon as I came across this issue that was opened around the same time as I encountered the problem.

I dug into the source code of the provider, but I did not think of testing the resource itself ( without using a module ) in my code.

I got some time to debug this thoroughly and that was when I identified the source of my problem as the module. I got that bug fixed and everything started working as expected.
